Don't hardcode shared libraries version number.

src/AudacityApp.cpp:845:11: error: expression is not assignable
   stderr = freopen("/dev/null", "w", stderr);

Index: src/AudacityApp.cpp
--- src/AudacityApp.cpp.orig
+++ src/AudacityApp.cpp
@@ -679,12 +679,12 @@ class GnomeShutdown
    {
       mArgv[0].reset(strdup("Audacity"));
 
-      mGnomeui = dlopen("libgnomeui-2.so.0", RTLD_NOW);
+      mGnomeui = dlopen("libgnomeui-2.so", RTLD_NOW);
       if (!mGnomeui) {
          return;
       }
 
-      mGnome = dlopen("libgnome-2.so.0", RTLD_NOW);
+      mGnome = dlopen("libgnome-2.so", RTLD_NOW);
       if (!mGnome) {
          return;
       }
@@ -841,8 +841,10 @@ int main(int argc, char *argv[])
    // never be able to get rid of the messages entirely, but we should
    // look into what's causing them, so allow them to show in Debug
    // builds.
+#ifndef __OpenBSD__
    stdout = freopen("/dev/null", "w", stdout);
    stderr = freopen("/dev/null", "w", stderr);
+#endif
 
    return wxEntry(argc, argv);
 }
